collector/meminfo_numa_linux.go

```diff
@@ -90,17 +97,31 @@ func getMemInfoNuma() (map[meminfoKey]float64, error) {
 		if err != nil {
 			return nil, err
 		}
-		for k, v := range numaInfo {
-			info[k] = v
+		metrics = append(metrics, numaInfo...)
+
+		file, err = os.Open(path.Join(node, "numastat"))
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
I can't see where you close the file descriptor again. Something along `defer numaStatFile.Close()` should be added here.